### PR TITLE
Rebase 87.0.4280.60

### PR DESCRIPTION
**Pull request by romandev**
Refs: #865
Co-authored-by: Byoungkwon Ko <codeimpl@gmail.com>
Co-authored-by: Jaeyong Bae <jdragon.bae@gmail.com>

### CODE REVIEW

File: src/org/triple/banana/toolbar/BottomToolbarController.java [hunk 1]
```diff
@@ -42,18 +40,16 @@
...
-    private OverviewModeBehavior mOverviewModeBehavior;
     private IncognitoStateProvider mIncognitoStateProvider;
     private boolean mIsIncognitoMode;
 
     @Override
-    public BananaBottomToolbarController init(View root, ActivityTabProvider tabProvider) {
-        // FIXME(#865): We should revisit the following code
+    public BananaBottomToolbarController init(
+            View root, ActivityTabProvider tabProvider, ThemeColorProvider themeColorProvider) {
         mViewGroup = new WeakReference<>(null);
```
Comment (jdragonbae) [inline src/org/triple/banana/toolbar/BottomToolbarController.java:51]: Can we remove it?

**Review by romandev — COMMENTED** [review]
Thank you for update. I did test with your patch, but the bottom toolbar is not shown when launching at first time.

**Comment by jdragonbae** [comment]
To show the bottom toolbar at first, can we set the default `true` at this code[1]?

[1] chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/CachedFeatureFlags.java:67

File: upstream/tb_chrome_android_java_src_org_chromium_chrome_browser_toolbar_bottom_BottomControlsCoordinator.java.patch [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
++                org.banana.cake.interfaces.BananaBottomToolbarController.get().init(
++                        root, tabProvider, themeColorProvider);
++
+         Toast.setGlobalExtraYOffset(
+                 root.getResources().getDimensionPixelSize(bottomControlsHeightId));
+ 
+         // Set the visibility of BottomControls to false by default. Components within
+         // BottomControls should update the visibility explicitly if needed.
+-        mMediator.setBottomControlsVisible(false);
++        mMediator.setBottomControlsVisible(true);
```
Comment (jdragonbae) [inline upstream/tb_chrome_android_java_src_org_chromium_chrome_browser_toolbar_bottom_BottomControlsCoordinator.java.patch:34]: Shouldn't we check the flag? `TabUiFeatureUtilities.is Tab Groups Android Enabled()`

**Comment by romandev** [comment]
> To show the bottom toolbar at first, can we set the default `true` at this code[1]?
> 
> [1] chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/CachedFeatureFlags.java:67

How the previous code works? I'm not sure but the previous code would be the same.

File: src/org/triple/banana/toolbar/BottomToolbarController.java (hunk 1 above)
Comment (codeimpl) [reply src/org/triple/banana/toolbar/BottomToolbarController.java:51]: sure~ thanks :)

**Comment by jdragonbae** [comment]
> > To show the bottom toolbar at first, can we set the default `true` at this code[1]?
> > [1] chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/CachedFeatureFlags.java:67
> 
> How the previous code works? I'm not sure but the previous code would be the same.

I think the previous code caches the `ChromeFeatureList.CHROME_DUET` flag when first run[1]. But now, related code seems to be removed.

[1] https://source.chromium.org/chromium/chromium/src/+/aa8f34fa4ff942fd100e55f5fe69b1fdf1a4ae1b:chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java;l=209

**Review by jdragonbae — APPROVED** [review]
LGTM

**Pull request merged**